src/main/java/mate/academy/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package mate.academy.service.shoppingcart.impl;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.cartitem.CartItemRequestDto;
+import mate.academy.dto.cartitem.CartItemUpdateRequestDto;
+import mate.academy.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.ShoppingCartMapper;
+import mate.academy.model.Book;
+import mate.academy.model.CartItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.repository.book.BookRepository;
+import mate.academy.repository.cartitem.CartItemRepository;
+import mate.academy.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.service.shoppingcart.ShoppingCartService;
+import mate.academy.service.user.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserService userService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
```

Comment:
suggest to add createShoppingCart(User user) method and call it in UserService register method
It's better to create shop cart for user during registration